src/services/dashboards/resourceFlat.test.js

```diff
@@ -0,0 +1,857 @@
+        activityReportObjectiveId: activityReportObjectiveDraft.id,
+        topicId: erseaTopicId,
+      },
+    });
+
+    arIds = [
+      reportOne.id,
+      reportTwo.id,
+      reportThree.id,
+      reportFour.id,
+      reportFive.id,
+      reportDraft.id,
+    ];
+  });
+
+  afterAll(async () => {
+    const reports = await ActivityReport
+      .findAll({ where: { userId: [mockUser.id] } });
+    const ids = reports.map((report) => report.id);
+    await NextStep.destroy({ where: { activityReportId: ids } });
+    await ActivityRecipient.destroy({ where: { activityReportId: ids } });
+
+    await ActivityReportObjectiveResource.destroy({
+      where: {
+        activityReportObjectiveId: activityReportOneObjectiveOne.id,
+      },
+    });
+    await ActivityReportObjectiveTopic.destroy({
+      where: {
+        activityReportObjectiveId: arIds,
+      },
+    });
+
+    // eslint-disable-next-line max-len
+    await ActivityReportObjective.destroy({ where: { objectiveId: [objective.id, objectiveTwo.id, objectiveThree.id] } });
+    await ActivityReport.destroy({ where: { id: ids } });
+    await Objective.destroy({ where: { id: [objective.id, objectiveTwo.id, objectiveThree.id] }, force: true });
+    await Goal.destroy({ where: { id: [goal.id, goalTwo.id, goalThree.id] }, force: true });
+    await Grant.destroy({ where: { id: GRANT_ID_ONE }, individualHooks: true });
+    await User.destroy({ where: { id: [mockUser.id] } });
+    await Recipient.destroy({ where: { id: RECIPIENT_ID } });
+    await db.sequelize.close();
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('resourceUseFlat', async () => {
+    const scopes = await filtersToScopes({ 'region.in': [REGION_ID], 'startDate.win': '2021/01/01-2021/01/31' });
+    let resourceUseResult;
+    db.sequelize.transaction(async () => {
+      ({ resourceUseResult } = await resourceFlatData(scopes));
+
+      expect(resourceUseResult).toBeDefined();
+      expect(resourceUseResult.length).toBe(3);
+
+      expect(resourceUseResult).toStrictEqual([
+        {
+          date: '2021-01-01',
+          url: 'https://eclkc.ohs.acf.hhs.gov/test',
+          rollUpDate: 'Jan-21',
+          title: null,
+          resourceCount: '2',
+          totalCount: '2',
+        },
+        {
+          date: '2021-01-01',
+          url: 'https://eclkc.ohs.acf.hhs.gov/test2',
+          rollUpDate: 'Jan-21',
+          title: null,
+          resourceCount: '3',
+          totalCount: '3',
+        },
+        {
+          date: '2021-01-01',
+          url: 'https://non.test1.gov/a/b/c',
+          rollUpDate: 'Jan-21',
+          title: null,
+          resourceCount: '2',
+          totalCount: '2',
+        },
+      ]);
+    });
+  });
+
+  it('resourceTopicUseFlat', async () => {
+    const scopes = await filtersToScopes({ 'region.in': [REGION_ID], 'startDate.win': '2021/01/01-2021/01/31' });
+    let topicUseResult;
+    db.sequelize.transaction(async () => {
```

Comment:
why are these in transactions?

Reply:
This is because the app will always be wrapped in a txn we need to add it for the test.